mimic/test/test_identity_objects.py

```diff
@@ -0,0 +1,302 @@
+
+    def test_listing_templates(self):
+        eeapi_template_id = 'some-template-id'
+        eeapi_template = ExampleEndpointTemplate(
+            name=self.eeapi_name,
+            uuid=eeapi_template_id,
+        )
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            endpoint_templates=[eeapi_template]
+        )
+
+        self.assertEqual(len(eeapi.list_templates()), 1)
+        eeapi.remove_template(eeapi_template_id)
+        self.assertEqual(len(eeapi.list_templates()), 0)
+
+    def test_invalid_endpoint_template(self):
+        """
+        Validate the endpoint template interface gate check
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        class InvalidTemplate(object):
+            pass
+
+        with self.assertRaises(TypeError):
+            eeapi.add_template(InvalidTemplate())
+
+    def test_duplicate_api_insertion_fails(self):
+        """
+        Validate only one template for by a given name (id) can
+        be added at a time.
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+        new_url = "https://api.new_region.example.com:9090"
+        new_region = "NEW_REGION"
+        new_eeapi_template_id = u"uuid-alternate-endpoint-template"
+        new_eeapi_template = ExampleEndpointTemplate(
+            name=self.eeapi_name,
+            uuid=new_eeapi_template_id,
+            region=new_region,
+            url=new_url
+        )
+
+        # first time succeeds
+        eeapi.add_template(new_eeapi_template)
+
+        # second time fails
+        with self.assertRaises(ValueError):
+            eeapi.add_template(new_eeapi_template)
+
+    def test_update_with_invalid_template(self):
+        class InvalidTemplate(object):
+            pass
+
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True,
+        )
+
+        with self.assertRaises(TypeError):
+            eeapi.update_template(InvalidTemplate())
+
+    def test_update_endpoint_template(self):
+        new_url = "https://api.new_region.example.com:9090"
+        new_region = "NEW_REGION"
+        new_eeapi_template_id = u"uuid-alternate-endpoint-template"
+        old_eeapi_template = ExampleEndpointTemplate(
+            name=self.eeapi_name,
+            uuid=new_eeapi_template_id,
+            region=new_region,
+        )
+        new_eeapi_template = ExampleEndpointTemplate(
+            name=self.eeapi_name,
+            uuid=new_eeapi_template_id,
+            region=new_region,
+            url=new_url
+        )
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True,
+            endpoint_templates=[old_eeapi_template]
+        )
+
+        eeapi.update_template(new_eeapi_template)
```

Comment:
Thanks for your explanation :).

Tests without assertions are a pattern I find difficult with which to work. I tend to view tests themselves as assertions about the system - the assertions in those tests are absolutely vital in communicating that the system will exhibit a certain behavior given the conditions that a test sets up. While it can be clear why an exception style test fails, it isn't immediately obvious what conditions should result in such a test passing.